### ISSUE TITLE
Dropdown menu autoFocus is not set correctly on menu reopening

### ISSUE THREAD

**Issue by Ke1sy** [ant-design/ant-design#53495]
### Reproduction link

[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/p/devbox/trigger-mode-antd-5-24-7-forked-5hwjgt?workspaceId=ws_QX6cCPfTB15EiJFJmZXTcE)

### Steps to reproduce

**Not selectable menu (works correctly for mouse users, but not keyboard users):**
1. Open the dropdown with keyboard (Enter).
2. Move focus to the 3rd menu item with keyboad (arrows up/down).
3. Close the dropdown with Escape.
4. Reopen the dropdown menu by clicking Enter.

*What is expected?*
Focus is set to the first focusable menu item.

*What is actually happening?*
Focus remains on the  last focused item (3rd menu item).

**Selectable menu (doesn't work correctly for mouse users and keyboard users).**

**Keyboard navigation:**
1. Open the dropdown with keyboard (Enter).
2. Move focus to the 3rd menu item with keyboad (arrows up/down) and select it (Enter).
3. Reopen the dropdown menu by clicking Enter.
4. Move focus to the 2nd menu item with keyboad (arrows up/down). Do not select it.
5. Close the dropdown with Escape.
4. Reopen the dropdown menu by clicking Enter.

*What is expected?*
Focus is set to the selected menu item (3rd menu item).

*What is actually happening?*
Focus remains on the last focused item (2nd menu item).

**Mouse navigation:**
1. Open the dropdown by mouse clicking.
2. Select 3rd menu item by clicking on it.
3. Reopen the dropdown menu.

*What is expected?*
Focus is set to the selected menu item (3rd menu item).

*What is actually happening?*
Focus is set to the 1st menu item.

### What is expected?

Focus is set correctly for all the described scenarios.

### What is actually happening?

Focus is not set correctly for all the described scenarios.

| Environment | Info |
| --- | --- |
| antd | 5.24.6 |
| React | 19.0.0 |
| System | Windows |
| Browser | any |

**Comment by xumengmvp** [comment]
You are so beautiful

**Comment by andrewhong04** [comment]
Can I work on this?

**Comment by afc163** [comment]
@andrewhong04 Sure.

**Comment by codewizardTM** [comment]
可以分配给我 @afc163

**Comment by codewizardTM** [comment]
Self-testing in progress
